apps/system/js/base_module.js

```diff
@@ -118,37 +119,60 @@
     _startSubModules: function() {
       if (!this.constructor.SUB_MODULES ||
           this.constructor.SUB_MODULES.length === 0) {
+        return Promise.resolve();
+      }
+
+      var submodules = this.constructor.SUB_MODULES.slice();
+      var unloaded = [];
+      submodules.forEach(function(submodule) {
+        if (BaseModule.defined(submodule) || window[submodule]) {
+          var name = BaseModule.lowerCapital(submodule);
+          if (!this[name]) {
+            this._initialSubModule(name, submodule);
+          }
+        } else {
+          unloaded.push(submodule);
+        }
+      }, this);
+
+      if (unloaded.length === 0) {
+        this.baseSubModuleLoaded && this.baseSubModuleLoaded();
         return;
       }
 
       this.debug('lazy loading submodules: ' +
-        this.constructor.SUB_MODULES.concat());
-      BaseModule.lazyLoad(this.constructor.SUB_MODULES).then(function() {
-        this.debug('lazy loaded submodules: ' +
-          this.constructor.SUB_MODULES.concat());
-        this.constructor.SUB_MODULES.forEach(function(module) {
-          var moduleName = BaseModule.lowerCapital(module);
-          var parent = this.constructor.SUB_MODULE_PARENT || this;
-          if (!parent[moduleName]) {
-            this._initialSubModule(moduleName, module);
-          }
-        }, this);
-      }.bind(this));
+        unloaded.concat());
+      return new Promise((resolve, reject) => {
```

Comment:
Remove this `new Promise()` block and simply return the promise from `Promise.all()` at L156.